### DISCUSSION TITLE
Jump next/prev only in the active file (and wrap)

### DISCUSSION THREAD

**Discussion by giallanon** [alefragnani/vscode-bookmarks#686] [Q&A] [answered]
Hello.
First of all congrats for your work, the extension is working very well.
I've search through options but did not find ont to create a shortcut to navigate throught bookmarsk only in the current file (maybe also with wrap around).
The idea is to have a shortcut that will cycle throught the bookmarks in current file and, when it reachs the last one, it goes back to the first one, but always in the same file.
Surely it's not a killer feature, but I'm so used to work like this that everytime the jump-next move me to another file my brain has a little shake :)

**Comment by alefragnani** [comment, marked as answer]
Hi @giallanon ,

First of all, thank you for your kind words. I’m glad to know the extension is being helpful.

About your request, there is no way to create shortcuts to this specific behavior. VS Code does not provide a way to use parameters/setting on shortcuts/commands. But, based on your comments, it seems this behavior is your preferred way to work, so I would suggest you to define the extension to work this way. To do so, simply use these two settings:


```json
    "bookmarks.navigateThroughAllFiles": false,
    "bookmarks.wrapNavigation": true
```

Based on the documentation:

> * Allow navigation through all files that contains bookmarks (true by default)
>     `"bookmarks.navigateThroughAllFiles": false`
> * Allow navigation to wrap around at the first and last bookmarks in scope (current file or all files) (true by default)
>     `"bookmarks.wrapNavigation": true`

Hope this helps

**Reply by giallanon** [reply]
Thank you @alefragnani , it works!
